python/cudf/cudf/core/dataframe.py

```diff
@@ -783,11 +783,23 @@ def fallback(col, fn):
                 else:
                     result[col] = fallback(rhs._cols[col], _reverse_op(fn))
         elif isinstance(other, Series):
-            raise NotImplementedError(
-                "Series to DataFrame arithmetic not supported "
-                "until strings can be used as indices. Try converting your"
-                " Series into a DataFrame first."
-            )
+            result_cols = list(self.columns)
+            for new_col in list(other.index):
```

Comment:
You shouldn't do `list(other.index)`, you should explicitly copy `other.index` to a list instead via `.tolist()` (needs to be plumbed into the Index classes. This can trigger a device to host copy element by element.